### PR TITLE
add external order and jwt defs

### PR DESCRIPTION
**Pull request by doodyparizada**

### CODE REVIEW

File: openapi.yaml [hunk 1]
```diff
@@ -87,6 +87,35 @@ paths:
...
+      - $ref: '#/components/parameters/RequestId'
+    post:
+      tags:
+        - Orders
+      summary: create an external order for a native offer
+      description: create an external order for a native offer
+      operationId: createExternalOrder
+      requestBody:
+        description: submission of an order
+        required: false # in case of spend the content is not required
```
Comment (doodyparizada) [inline openapi.yaml]: @nitzantomer noted this should be true

File: openapi.yaml [hunk 2]
```diff
@@ -478,6 +525,15 @@ components:
...
+    ExternalOrderRequest:
+      type: object
+      required:
+        - jwt
+      properties:
+        jwt:
+          type: string
+          description: string serialized JWTBodySpend
+
     OpenOrder:
```
Comment (YohayBar) [inline openapi.yaml]: need to add to openOrder amount and recipient wallet address

File: openapi.yaml [hunk 3]
```diff
@@ -305,19 +334,37 @@ components:
...
+              description: a coupon code - appears in a result object of an Order
+              type: string
+              example: aaaa-bbbb-cccc
+
+    JWTBodyConfirmPaymentResult:
+      allOf:
+        - $ref: '#/components/schemas/OrderSpendResult'
+        - type: object
+          required:
+            - jwt
```
Comment (YohayBar) [inline openapi.yaml]: add doc explaining what this JWT contain

File: openapi.yaml [hunk 4]
```diff
@@ -305,19 +334,37 @@ components:
...
       properties:
-        coupon_code:
-          description: "a coupon code - appears in a result object of an Order"
-          type: string
-          example: aaaa-bbbb-cccc
         type:
           type: string
-          enum:
-            - coupon
+          enum: [coupon, payment]
```
Comment (YohayBar) [inline openapi.yaml]: add discriminator to the type

File: openapi.yaml (hunk 3 above)
Comment (doodyparizada) [reply openapi.yaml]: It has 2 lines down

File: openapi.yaml (hunk 2 above)
Comment (doodyparizada) [reply openapi.yaml]: 👍

**Pull request merged**